src/main/java/org/dice/factcheckbenchmark/benchmark/rdf/RDFResource.java

```diff
@@ -0,0 +1,85 @@
+package org.dice.factcheckbenchmark.benchmark.rdf;
+
+import org.apache.jena.rdf.model.*;
+
+import java.util.*;
+
+/**
+ * @author DANISH AHMED on 3/22/2018
+ */
+public class RDFResource {
+
+    protected Resource resource;
+    protected Model model;
+    protected String label;
+    public String uri;
+
+    private Property owlSameAsProperty = ResourceFactory.createProperty(Constants.OWL_NAMESPACE + "sameAs");
+
+    public List<Resource> owlSameAsList = new ArrayList<Resource>();     // only considering english for now
+    public Map<String, String> langLabelsMap = new HashMap<String, String>();
+
+    /**
+     * give me FactCheck resource model
+     * @param resource given jena resource
+     * @param model jena model
+     */
+    RDFResource(Resource resource, Model model) {
+        this.resource = resource;
+        this.uri = resource.getURI();
+        this.model = model;
+
+        // set labels w.r.t language
+        Property labelProperty = ResourceFactory.createProperty(Constants.RDF_SCHEMA_NAMESPACE + "label");
+        NodeIterator labelNodeIterator = this.model.listObjectsOfProperty(this.resource, labelProperty);
+
+        // set sameAs resource list
+        getResourceLabel(labelNodeIterator);
+        setOwlSameAsList();
+    }
+
+    private void getResourceLabel(NodeIterator nodeIterator) {
+        while (nodeIterator.hasNext()) {
+            RDFNode rdfNode = nodeIterator.nextNode();
+            String lang = rdfNode.asLiteral().getLanguage();
+            String label = rdfNode.asLiteral().getLexicalForm();
+
+            langLabelsMap.put(lang, label);
+            if (lang.equals("en")) {
+                this.label = label;
+            }
+        }
+    }
+
+    /**
+     * get sameAs owl property
+     */
+    private void setOwlSameAsList() {
+        NodeIterator nodeIterator = model.listObjectsOfProperty(this.resource, this.owlSameAsProperty);
+        while (nodeIterator.hasNext()) {
+            RDFNode rdfNode = nodeIterator.nextNode();
+            owlSameAsList.add(rdfNode.asResource());
```

Comment:
I have updated the code because the user can trick the system by placing literal against sameAs. In general, this error (for owl:sameAs) will never occur, as owl:sameAs always points to a resource. It can be tried by running this query which returns 0 rows:

PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX dbpediaowl: <http://dbpedia.org/ontology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
select distinct ?o where {
  ?s owl:sameAs ?o.
  Filter (isLiteral(?o))
}
limit 100

Reply:
In clean and correct datasets, it never occurs. However, not all datasets are clean and correct :disappointed: